service/history/historyEngine.go

```diff
@@ -3186,3 +3237,7 @@ func (e *historyEngineImpl) loadWorkflow(
 
 	return nil, serviceerror.NewInternal("unable to locate current workflow execution")
 }
+
+func (e *historyEngineImpl) metricsScope(ctx context.Context) metrics.Scope {
+	return interceptor.MetricsScope(ctx, e.logger)
+}
```

Comment:
This is something wrong. `interceptor` is gRPC interceptor which deals with metrics on RPC side. It can't be used in `historyEngine`. I think you need something like:
```
e.metricsClient.Scope(...)
```
to get a scope.

Reply:
That's what I did at first. But the goal was to keep the metrics looking the same as much as possible, and the frontend uses the rpc interceptor metrics (which include the method name and namespace). This function is eventually getting called from the history service grpc handler, right? (Not necessarily in tests, but it doesn't matter in tests.)

Reply:
Hmm, maybe it's cleaner to have the rpc handler get a metrics scope and pass it down to historyEngine? Or would you suggest not using the rpc interceptor scope at all? That'll change the tags on `metrics.EventBlobSize` for these checks.

Reply:
> This is something wrong. interceptor is gRPC interceptor which deals with metrics on RPC side

most of the case yes, but ctx is also initialized with the metrics scope so if business logic need to use it, business logic can

Reply:
It looks more natural in `workflowHandler` because we know that all request to the handler went through interceptor. In `historyEngine` it is not that obvious. I think `MetricsScope` should be moved from `interceptor` package to something metrics specific and should be called `metrics.ScopeFromContext`. But this is out of scope for this PR.

Reply:
I agree it makes more sense to call `interceptor.MetricsScope` from an rpc handler than from `historyEngine`. The problem is the lifetime of `historyEngine` extends beyond a single handler function's scope, so we'd need to pass in the `metrics.Scope` with every call where it's required. At that point it makes more sense to bundle it into the context... and it's already in the context, so we can just use it from there.

One thing we could do in this PR is to move the size checks into the history rpc handler instead of the engine. It doesn't look like that would miss any call paths, and there's enough information to do the checks there. Then it would be natural to use the metrics scope from the interceptor context.